### DISCUSSION TITLE
Issue installing iraf-2.16.1-2021.06.14 on Mac running MacOS 10.15

### DISCUSSION THREAD

**Discussion by goudfroo** [iraf-community/iraf#201] [Q&A] [answered]
**Describe the bug**
Hello there,

I'm trying to install iraf-2.16.1-2021.06.14 on my Mac running MacOS 10.15 (Catalina). Following the INSTALL.md file, something goes wrong in the step
    $ make sysgen 2>&1 | tee build.log

The error occurs in the "GENERIC" package, see below:
+ echo '----------------------- GENERIC ------------------------'
----------------------- GENERIC ------------------------
+ cd generic
+ sh -x mkpkg.sh
+ set -e
+ flex -o lexyy.c tok.l
+ gcc -c -g -Wall -O2 -I/Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/include generic.c yywrap.c lexyy.c
+ gcc generic.o lexyy.o yywrap.o /Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/unix/hlib/libboot.a /Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/lib/libsys.a /Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/lib/libvops.a /Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/unix/hlib/libos.a /Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/unix/bin.macintel/libf2c.a -lm -o generic.e
ld: file too small (length=8) file '/Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/lib/libsys.a' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [sysgen] Error 1

Can someone give me a hint on how to resolve this? I'm attaching the build.log file.

Paul Goudfrooij, goudfroo@stsci.edu

[build.log](https://github.com/iraf-community/iraf/files/7486932/build.log)
.

**Comment by olebole** [comment]
Did you use the git, the tarball or the zip file? The zip file doesn't do the symbolic links correct; you should use the tarball.
If this is not the case, can you check:
```
$ ls -l lib/libsys.a
lrwxrwxrwx 1 oles oles 15 14. Jun 09:20 lib/libsys.a -> ../bin/libsys.a
```
This should really be a symbolic link. If this is not the case, please check that you unpacked it correctly, Otherwise, can you do a `cat lib/libsys.a` to evaluate what is actually in the file?

**Comment by goudfroo** [comment]
Hi Ole, 

thanks for your rapid response. I used the tar file, and yes, the file in question is a symbolic link. Here is what’s in the libsys.a file:

wazoo:~/Installs/iraf-2.16.1-2021.06.14/lib > ls -l libsys.a 
lrwxrwx---  1 goudfroo  ststaff  15 Jun 14 02:33 libsys.a@ -> ../bin/libsys.a
wazoo:~/Installs/iraf-2.16.1-2021.06.14/lib > cat libsys.a 
!<arch>
wazoo:~/Installs/iraf-2.16.1-2021.06.14/lib > 

Regards,
Paul

----
Paul Goudfrooij
Observatory Scientist
JWST/NIRISS Deputy Branch Manager and Operations Lead
Space Telescope Science Institute
3700 San Martin Drive 
Baltimore, MD 21218, USA
My Website

-------------------------------------------------
Safe Zone Member
web.jhu.edu/SafeZone

**Reply by olebole** [reply]
This is strange. From the log, there seems the "xc" command not found:
```
check out file `libsys.a = /Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/lib/libsys.a'
warning: library `libsys.a' not found
xc -c -w access.x aread.x areadb.x await.x awaitb.x awrite.x awriteb.x close.x delete.x deletefg.x diropen.x falloc.x fcache.x …
xc: Command not found.
ar r /Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/lib/../bin/libsys.a access.o aread.o areadb.o await.o awaitb.o awrite.o …
ar: creating archive /Users/goudfroo/Installs/iraf-2.16.1-2021.06.14/lib/../bin/libsys.a
ar: access.o: No such file or directory
ar: aread.o: No such file or directory
…
```
Could it be that xc points to some previous incomplete or removed installation of IRAF? In principle, you should have the following link (created with `./install`):
```
$ ls -l ~/.iraf/bin/xc
lrwxrwxrwx 1 oles oles 65 25. Okt 09:45 /home/oles/.iraf/bin/xc -> /home/oles/Projects/iraf/iraf/unix/bin.linux64/xc.e
```
(mine is a Linux; your should point to the path to your new installation, and `bin.macintel`).

If yes, check that `bin.macintel/xc.e` exists and is executable; your log however shows that it was created successfully.

**Reply by Dipankar-13** [reply]
i have the same link as mentioned in my mac m1 but i am not being able to run iraf from terminal with cl command. Every-time i have to go to the .iraf/bin directory then give ./cl command then only it runs iraf. Can you please provide a solution??

**Reply by olebole** [reply]
Doesn't it work to just add this directory to your PATH?

**Comment by goudfroo** [comment, marked as answer]
Thanks, Ole. 

never mind - there must have been something in my $PATH that caused the issue. I started a new clean shell from scratch and then it worked. Sorry for the trouble and thanks for your assistance!

Cheers, 
Paul

----
Paul Goudfrooij
Observatory Scientist
JWST/NIRISS Deputy Branch Manager and Operations Lead
Space Telescope Science Institute
3700 San Martin Drive 
Baltimore, MD 21218, USA
My Website

-------------------------------------------------
Safe Zone Member
web.jhu.edu/SafeZone
